packages/strapi/lib/core/configurations.js

```diff
@@ -318,9 +318,30 @@ module.exports.app = async function() {
     return acc;
   }, {});
 
+  // default settings
   this.config.port = get(this.config.currentEnvironment, 'server.port') || this.config.port;
   this.config.host = get(this.config.currentEnvironment, 'server.host') || this.config.host;
-  this.config.url = `http://${this.config.host}:${this.config.port}`;
+
+  // default construct url
+  this.config.url = `http://${this.config.host}:${this.config.port}`
+
+  // proxy settings
```

Comment:
`this.config.proxy` not exist, all settings are not mount in `strapi.config`

So in this case you can just have this line:
`this.config.proxy = get(this.config.currentEnvironment, 'server.proxy' || {}); `

Reply:
@lauriejim I defined the this.config.proxy a few lines down: `this.config.proxy = this.config.currentEnvironment.server.proxy;`

Reply:
Modified to match your example